trillian/migrillian/core/controller.go

```diff
@@ -259,13 +260,23 @@ func (c *Controller) Run(ctx context.Context) error {
 		}()
 	}
 
+	if c.opts.ResignDelay != 0 { // Configured with mastership resignation.
```

Comment:
It feels a bit like this would belong better in `RunWhenMaster`, so `Run` could continue to be unaware of mastership elections and just do what it says on the tin,  Is it possible to move this out and cancel the passed-in context after an resignation interval?

Reply:
It's possible, yes. Doing it here though is a bit nicer because it allows graceful termination, so that the data stored in Trillian doesn't have gaps, and a follow-up `Fetcher` run does fewer redundant downloads.

Reply:
Another option is to split `Run` into `Start` and `Stop`, and call `Stop` from `RunWhenMaster`.

Reply:
Leaving it here is probably OK, but I'd change the name/comments -- rather than being `ResignDelay` it could be something like `StopAfter` or `FetchDuration` so there's no need to mention mastership.

Reply:
Done.